assert/assertion_compare_test.go

```diff
@@ -1,10 +1,26 @@
 package assert
 
 import (
+	"bytes"
+	"fmt"
 	"reflect"
 	"testing"
 )
 
+type outputT struct {
+	buf *bytes.Buffer
+}
+
+// Implements TestingT
+func (t *outputT) Errorf(format string, args ...interface{}) {
+	s := fmt.Sprintf(format, args...)
+	t.buf.WriteString(s)
+}
+
+func (t *outputT) get() string {
+	return string(t.buf.Bytes())
+}
+
```

Comment:
Given that we only use this logic for the tests you just added, do you mind moving it closer to them?